components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -532,8 +542,45 @@ std::vector<mojom::FeedItemV2Ptr> GenerateBlockFromContentGroups(
       },
       std::move(get_weighting));
 
+  base::flat_set<std::string> politics_publisher_ids;
+  for (const auto& [publisher_id, publisher] : publishers) {
+    auto channels = GetChannelsForPublisher(locale, publisher);
+    if (base::Contains(channels, "Politics")) {
+      politics_publisher_ids.insert(publisher_id);
+    }
+  }
+
+  auto get_discovery_weighting = base::BindRepeating(
+      [](base::flat_set<std::string> politics_publisher_ids) {
+        return base::BindRepeating(
+            [](base::flat_set<std::string> politics_publisher_ids,
+               const mojom::FeedItemMetadataPtr& metadata,
+               const ArticleWeight& weight) {
+              if (weight.subscribed) {
+                return 0.0;
+              }
+
+              auto is_politics =
+                  politics_publisher_ids.find(metadata->publisher_id);
+              if (is_politics != politics_publisher_ids.end()) {
```

Comment:
Maybe instead of doing it this way, we should add an `is_discoverable` property to the `ArticleWeight` struct and calculate it as part of `GetArticleWeight`

Reply:
Great point 💯